ui/js/controllers/perf/compare.js

```diff
@@ -307,76 +388,95 @@ perf.controller('CompareResultsCtrl', [
+                                        parseInt($stateParams.showOnlyImportant)),
                 showOnlyConfident: Boolean($stateParams.showOnlyConfident !== undefined &&
                                            parseInt($stateParams.showOnlyConfident))
             };
 
-            function updateURL() {
-                $state.transitionTo('compare', {
-                    framework: $scope.filterOptions.framework.id,
-                    filter: $scope.filterOptions.filter,
-                    showOnlyImportant: $scope.filterOptions.showOnlyImportant ? undefined : 0,
-                    showOnlyConfident: $scope.filterOptions.showOnlyConfident ? 1 : undefined
-                }, {
-                    location: true,
-                    inherit: true,
-                    relative: $state.$current,
-                    notify: false
-                });
-            }
-
-
             $scope.originalProject = ThRepositoryModel.getRepo(
                 $stateParams.originalProject);
             $scope.newProject = ThRepositoryModel.getRepo(
                 $stateParams.newProject);
             $scope.newRevision = $stateParams.newRevision;
-            $scope.originalRevision = $stateParams.originalRevision;
-
-            verifyRevision($scope.originalProject, $scope.originalRevision, "original").then(function () {
-                verifyRevision($scope.newProject, $scope.newRevision, "new").then(function () {
-                    if ($scope.errors.length > 0) {
-                        $scope.dataLoading = false;
-                        return;
-                    }
-                    $scope.$watchGroup(['filterOptions.filter',
-                                        'filterOptions.showOnlyImportant',
-                                        'filterOptions.showOnlyConfident'],
-                                       updateURL);
-
-                    $scope.$watch('filterOptions.framework',
-                                  function(newValue, oldValue) {
-                                      if (newValue.id !== oldValue.id) {
-                                          updateURL();
-                                          load();
-                                      }
-                                  });
-                    load();
+            if($stateParams.originalRevision !== undefined){
+                $scope.originalRevision = $stateParams.originalRevision;
+                verifyRevision($scope.originalProject, $scope.originalRevision, "original");
+            }
+            if($stateParams.originalRevision === undefined){
+                $scope.timeRanges = phTimeRanges;
+                $scope.selectedTimeRange = _.find($scope.timeRanges, {
+                    value: ($stateParams.selectedTimeRange) ? parseInt($stateParams.selectedTimeRange) : compareBaseLineDefaultTimeRange
                 });
+                $scope.timeRangeChanged = function(selectedTimeRange){
```

Comment:
I would define this function at a higher level of indentation, maybe just after the updateURL() definition. Also it might be helpful adding a short comment saying what it is for.